rest-service/manager_rest/cluster_status_manager.py

```diff
@@ -0,0 +1,237 @@
+#########
+# Copyright (c) 2019 Cloudify Platform Ltd. All rights reserved
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+#  * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  * See the License for the specific language governing permissions and
+#  * limitations under the License.
+
+import copy
+import json
+from os import path
+from datetime import datetime, timedelta
+
+from cloudify.cloudify_nodes import (ServiceType,
+                                     ServiceStatus,
+                                     NodeServiceStatus)
+
+from manager_rest.storage import models, get_storage_manager
+from manager_rest.rest.rest_utils import parse_datetime_string
+
+
+DB_SERVICE_KEY = 'PostgreSQL'
+BROKER_SERVICE_KEY = 'RabbitMQ'
+UNINITIALIZED_STATUS = 'Uninitialized'
+CLUSTER_STATUS_PATH = '/opt/manager/resources/cluster_status'
+
+
+class ClusterStatusManager(object):
```

Comment:
in general i don't see a reason for this to be a class, and passing through things between methods as attributes on an instance (and there is ever only one instance at a time anyway)

i think this really wants to just be a function, and all its methods would be just functions too, and they would simply take arguments instead of accessing instance attributes

the problem is that this is harder to read because looking at a method eg. `_handle_missing_non_manager_files`, it accesses `self.missing_status_files`, and now i need to examine the rest of the class to find where is `.missing_status_files` set, instead of just looking at the caller

so all in all i prefer more structure, by passing arguments, instead of a free-for-all accessing anything anywhere

Reply:
I agree, in the beginning I thought I'll use more the instance variables but now I see you are right and I changed it to be just a function.